pkg/cluster/provider.go

```diff
@@ -68,20 +69,54 @@ func NewProvider(options ...ProviderOption) *Provider {
 		}
 	}
 
+	// ensure a provider if none was set
+	// NOTE: depends on logger being set (see sorting above)
 	if p.provider == nil {
-		// auto-detect based on each package IsAvailable() function
-		// default to docker for backwards compatibility
-		if docker.IsAvailable() {
-			p.provider = docker.NewProvider(p.logger)
-		} else if podman.IsAvailable() {
-			p.provider = podman.NewProvider(p.logger)
-		} else {
-			p.provider = docker.NewProvider(p.logger)
+		// DetectNodeProvider does not fallback to allow callers to determine
+		// this behavior
+		// However for compatibility if the caller of NewProvider supplied no
+		// option and we autodetect internally, we default to the docker provider
+		// for fallback, to avoid a breaking change for now.
+		// This may change in the future.
+		// TODO: consider breaking this API for earlier errors.
+		providerOpt, _ := DetectNodeProvider()
+		if providerOpt == nil {
+			providerOpt = ProviderWithDocker()
 		}
+		providerOpt.apply(p)
 	}
 	return p
 }
 
+// NoNodeProviderDetectedError indicates that we could not autolocate an available
+// NodeProvider backend on the host
+var NoNodeProviderDetectedError = errors.NewWithoutStack("failed to detect any supported node provider")
```

Comment:
we haven't actually had much in the way of client-handle-able errors yet (usually stuff like "cluster bringup failed in some unexpected way"), so I had to update the error utilities to handle this appropriately.